sycl/doc/extensions/experimental/sycl_ext_oneapi_complex.asciidoc

```diff
@@ -0,0 +1,384 @@
+    template <class T> complex<T> pow(const complex<T> &, const complex<T> &);
+    /// Compute complex number z raised to the power of complex number y.
+    template <class T, class U> complex</*Promoted*/> pow(const complex<T> &, const complex<U> &);
+    /// Compute complex number z raised to the power of real number y.
+    template <class T, class U> complex</*Promoted*/> pow(const complex<T> &, const U &);
+    /// Compute real number x raised to the power of complex number y.
+    template <class T, class U> complex</*Promoted*/> pow(const T &, const complex<U> &);
+
+    /// Compute the inverse hyperbolic sine of complex number x.
+    template <class T> complex<T> asinh(const complex<T> &);
+    /// Compute the inverse hyperbolic cosine of complex number x.
+    template <class T> complex<T> acosh(const complex<T> &);
+    /// Compute the inverse hyperbolic tangent of complex number x.
+    template <class T> complex<T> atanh(const complex<T> &);
+    /// Compute the hyperbolic sine of complex number x.
+    template <class T> complex<T> sinh(const complex<T> &);
+    /// Compute the hyperbolic cosine of complex number x.
+    template <class T> complex<T> cosh(const complex<T> &);
+    /// Compute the hyperbolic tangent of complex number x.
+    template <class T> complex<T> tanh(const complex<T> &);
+    /// Compute the inverse sine of complex number x.
+    template <class T> complex<T> asin(const complex<T> &);
+    /// Compute the inverse cosine of complex number x.
+    template <class T> complex<T> acos(const complex<T> &);
+    /// Compute the inverse tangent of complex number x.
+    template <class T> complex<T> atan(const complex<T> &);
+    /// Compute the sine of complex number x.
+    template <class T> complex<T> sin(const complex<T> &);
+    /// Compute the cosine of complex number x.
+    template <class T> complex<T> cos(const complex<T> &);
+    // Compute the tangent of complex number x.
+    template <class T> complex<T> tan(const complex<T> &);
+
+} // namespace experimental
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+== Suffixes for complex number literals
+
+This proposal describes literal suffixes for constructing complex number
+literals.
+The suffixes `i` and `if` create complex numbers of the types `complex<double>`
+and `complex<float>` respectively, with their imaginary part denoted by the
+given literal number and the real part being zero.
+
+```C++
+namespace sycl {
+namespace ext {
+namespace oneapi {
+
+namespace literals {
+namespace complex_literals {
+constexpr complex<double> operator""i(long double x);
+constexpr complex<double> operator""i(unsigned long long x);
+
+constexpr complex<float> operator""if(long double x);
+constexpr complex<float> operator""if(unsigned long long x);
+} // namespace complex_literals
+} // namespace literals
+
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
```

Comment:
Given my comments above, I'm not sure it's useful to add these literals.  If you do keep it, you should add a statement to the spec saying that the application must add a `using namespace` statement like this in order to gain access to these literals:

```
using namespace sycl::ext::oneapi::experimental::literals::complex_literals;
```

Also, note that your literal suffixes "i" and "if" are the same as the C++ standard suffixes.  Therefore, I think an application would not be able to enable both at the same time.  This might make it hard for an application to use both `std::complex` and `sycl::ext::oneapi::experimental::complex` in the same source file.

Finally, it seems like you did not add a literal for half-precision complex literals.  Was this an oversight?

Reply:
@abagusetty

Reply:
I am more inclined to add the support for float literals since it would help few libraries (oneDPL) to adapt the sycl complex spec instead of currently relying on std complex on the device-side which exposed a host of issues for portability. 

Can add text to explicitly use `using namespace sycl::ext::oneapi::experimental::literals::complex_literals;`

Thanks for catching the half-precision.

Reply:
@jle-quel if you feel comfortable, please feel free to make the necessary changes (once I get some feed back) for a shorter turn-around. Or I could create a PR to your branch.

Reply:
As @gmlueck mentioned, I don't think you can change the arguments to double as a short-term solution either. I believe that you should be making the necessary changes since you are the one that introduced it and knows best how to defend your implementation and extension proposal @abagusetty